src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,225 @@
+                         Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create car: " + car + ".", e);
+        }
+        if (car.getId() != null) {
+            addDrivers(car.getId(), car.getDrivers());
+        }
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(final Long carId) {
+        Optional<Car> carResult = Optional.empty();
+        String query = "SELECT c.id AS car_id, model AS car_model, manufacturer_id AS m_id, "
+                + "name AS m_name, country AS m_country FROM cars c "
+                + "JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.id "
+                + "WHERE c.id = ? AND c.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, carId);
+            statement.executeQuery();
+            ResultSet resultSet = statement.getResultSet();
+            if (resultSet.next()) {
+                Car car = new Car();
+                car.setId(carId);
+                car.setModel(resultSet.getString(2));
+                car.setManufacturer(generateManufacturer(resultSet));
+                carResult = Optional.of(car);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get car by id: " + carId + ".", e);
+        }
+        carResult.ifPresent(car -> car.setDrivers(getAllDriversByCarId(carId)));
+        return carResult;
+    }
+
+    @Override
+    public List<Car> getAll() {
+        List<Car> cars = new ArrayList<>();
+        String query = "SELECT cars.id, cars.model, cars.manufacturer_id, "
+                + "manufacturers.name, manufacturers.country "
+                + "FROM cars "
+                + "INNER JOIN manufacturers ON cars.manufacturer_id = manufacturers.id "
+                + "WHERE cars.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.executeQuery();
+            ResultSet resultSet = statement.getResultSet();
+            while (resultSet.next()) {
+                cars.add(generateCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get all cars from DB.", e);
+        }
+        if (cars.size() > 0) {
+            cars.forEach(c -> c.setDrivers(getAllDriversByCarId(c.getId())));
+        }
```

Comment:
Do we need this check? Do you think foreach will work on an empty list?

Reply:
Done